### PR TITLE
StatsD and Datadog

### LINKED ISSUE #646: Using datadog with Cloud

### PR DESCRIPTION
**Pull request by begriffs**
Fixes #646

### CODE REVIEW

File: cloud/monitoring.rst [hunk 1]
```diff
@@ -80,6 +80,120 @@ Failovers happen to address hardware failure, as mentioned, and also for other r
 *  Situation
 *  Follower
 
+StatsD external reporting
+=========================
+
+Citus Cloud can send events to an external `StatsD <https://github.com/etsy/statsd>`_ server for detailed monitoring. Cloud sends the following statsd metrics:
```
Comment (lfittl) [inline cloud/monitoring.rst]: I would say “Citus Cloud” instead of just “Cloud” towards the end of that sentence

File: cloud/monitoring.rst [hunk 2]
```diff
@@ -80,6 +80,120 @@ Failovers happen to address hardware failure, as mentioned, and also for other r
...
+| citus.locks.granted.<mode>.<locktype>.count | See below                          |
++---------------------------------------------+------------------------------------+
+| citus.mem.available                         |                                    |
++---------------------------------------------+------------------------------------+
+| citus.mem.buffered                          |                                    |
++---------------------------------------------+------------------------------------+
+| citus.mem.cached                            |                                    |
++---------------------------------------------+------------------------------------+
+| citus.mem.commit_limit                      | Memory currently available to      |
+|                                             | be allocated on the system         |
```
Comment (lfittl) [inline cloud/monitoring.rst]: Might be worth mentioning that these come from /proc/meminfo, so if one wants to understand more they can look there.

File: cloud/monitoring.rst [hunk 3]
```diff
@@ -80,6 +80,120 @@ Failovers happen to address hardware failure, as mentioned, and also for other r
...
++---------------------------------------------+------------------------------------+
+| citus.pgbouncer_outbound.cl_waiting         | Waiting client connections         |
++---------------------------------------------+------------------------------------+
+| citus.pgbouncer_outbound.sv_active          | Active server connections          |
++---------------------------------------------+------------------------------------+
+| citus.pgbouncer_outbound.sv_idle            | Idle server connections            |
++---------------------------------------------+------------------------------------+
+| citus.pgbouncer_outbound.sv_used            | Server connections idle more       |
+|                                             | than server_check_delay            |
++---------------------------------------------+------------------------------------+
```
Comment (lfittl) [inline cloud/monitoring.rst]: I’m not fully sure that description is correct - is that straight from the pgbouncer docs?

Comment (begriffs) [reply cloud/monitoring.rst]: I shortened the description slightly, but yes. https://pgbouncer.github.io/usage.html

File: cloud/monitoring.rst (hunk 2 above)
Comment (begriffs) [reply cloud/monitoring.rst]: Added a note about that below this table.

**Pull request merged**